doc/advanced-configuration/auth.md

```diff
@@ -0,0 +1,134 @@
+
+### `auth.sasl_external`
+* **Syntax:** list of strings, allowed values: `"standard"`, `"common_name"`, `"auth_id"`
+* **Default:** `["standard"]`
+* **Example:** `sasl_external = ["standard", "common_name"]`
+
+There are three possible ways of using the `SASL EXTERNAL` mechanism:
+
+* `standard` - do not accept a certificate with no `xmpp_addrs` field (default),
+* `common_name` - use the `common_name` field if it is provided in the certificate,
+* `auth_id` - accept a certificate without `xmpp_addrs` and use the user identity from the authentication request.
+
+This option allows you to list the enabled ones in the order of preference (they are tried until one succeeds or the list is exhausted).
+
+## Password-related options
+
+These options are common to the `http`, `rdbms`, `internal` and `riak` methods.
+
+### `auth.password.format`
+* **Syntax:** string, one of: `"plain"`, `"scram"`
+* **Default:** `"scram"`
+* **Example:** `password.format = "plain"`
+
+Decide whether user passwords will be kept plain or hashed in the database.
+Currently, popular XMPP clients support the SCRAM method and it is strongly recommended to use the hashed version.
+The older XMPP clients can still use the `PLAIN` mechanism.
+
+**Note:** The `DIGEST-MD5` mechanism is not available with the `scram` password format.
+
+### `auth.password.hash`
+* **Syntax:** list of strings, allowed values: `"sha"`, `"sha224"`, `"sha256"`, `"sha384"`, `"sha512"`
+* **Default:** not set - all hash functions supported
+* **Example:** `password.hash = ["sha384", "sha512"]`
+
+MongooseIM supports SHA-1, SHA-224, SHA-256, SHA-384 and SHA-512 for SCRAM hashing.
+You can use this option to limit the supported hash functions by listing them explicitly.
+The value `"sha"` stands for the SHA-1 algorithm.
+
+### `auth.scram_iterations`
+* **Syntax:** positive integer
+* **Default:** 10000,  as recommended in this [XEP](https://xmpp.org/extensions/xep-0438.html#pbkdf2) and this [NIST Guidelines](https://pages.nist.gov/800-63-3/sp800-63b.html#sec5)
+* **Example:** `scram_iterations = 20_000`
+
+Hash function round count.
+This is a tradeoff between latency and security.
+The higher the value, the more difficult breaking the hashes is: increasing the count increases the work it requires to compute a full derivation, which effectively slows down brute-force attacks.
+But it adds load on both client and server, so this parameter should be tuned as high as the business-rules allow.
+Note that increasing the security of a password has a higher impact over the security of the algorithm, without impacting its load.
+See more information in this [NIST guide, Appendix A.2.2](https://csrc.nist.gov/publications/detail/sp/800-132/final)
+
+## Example
+
+The minimal authentication setup with password verification uses the internal Mnesia database:
+
+```toml
+[auth]
+  methods = ["internal"]
+```
+
+The supported SASL mechanisms `PLAIN`, `SCRAM-*` and `DIGEST-MD5` will be enabled by default.
```

Comment:
What exactly does it mean that `SCRAM-*` will be enabled by default? All of them?

Reply:
All of them, I reworded this fragment and linked to capabilities. Maybe it's cleaner now where this list comes from.